### DISCUSSION TITLE
Best way to use joined fields? (for now)

### DISCUSSION THREAD

**Discussion by Lord-Platypus** [serverpod/serverpod#2764] [Q&A] [answered]
I know that joined fields are not a thing yet, but...

I have three tables: Items, Languages and ItemTranslations with name and description fields.
When I get the items, I want to use name and description like this: item.name.
This is my model for now

```
class: Item
table: items
fields:
  translations: ItemTranslation?, relation
```
So, when I want to use the name field I need to go through "translations"

`item.translations.name`

For now my solution is using an extension method and doing something like this:

```
extension ItemExtension on Item {
    String get name => translations.name;
}
```
But like this I will have to remember that there are some fields that are getters and are not in the model. Not a big deal, but I would prefer to have everything in one place.

Do you have some ideas? Or is this the best way to achieve this?

**Comment by SandPod** [comment, marked as answer]
This is currently the best way to achieve this.

If you want `ItemTranslation` to be stored in a shared table this is the way to go in the future as well.

With inheritance, https://github.com/serverpod/serverpod/issues/1873, it will be possible to have field that are shared between multiple models. But these fields will then need to be save in tables connected to each different model and not in a shared table.
